File: module-server/src/main/java/de/twomartens/wahlrecht/model/internal/SeatResult.java
package de.twomartens.wahlrecht.model.internal;

import java.util.LinkedList;
import java.util.Map;
import java.util.Map.Entry;
import java.util.stream.Collectors;
import lombok.Builder;

@Builder
public record SeatResult(Map<VotingResult, Integer> seatsPerResult,
                         LinkedList<Double> usedElectionNumbers) {

  public Map<Nomination, Integer> seatsPerNomination() {
    return seatsPerResult.entrySet().stream()
        .collect(Collectors.toMap(entry -> entry.getKey().getNomination(), Entry::getValue));
  }
}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/model/internal/SeatResult.kt
package de.twomartens.wahlrecht.model.internal

data class SeatResult(val seatsPerResult: Map<VotingResult, Int>,
                      val usedElectionNumbers: List<Double>) {
    val seatsPerNomination = seatsPerResult.entries.asSequence()
        .map {Pair(it.key.nominationId, it.value)}
        .toMap()
}
